lib/widgets/screens/messages/messages_list.dart

```diff
@@ -1,14 +1,14 @@
 import 'package:flutter/material.dart';
+import 'package:mm_flutter_app/__generated/schema/operations.graphql.dart';
 import 'package:mm_flutter_app/data/models/messages/unseen_messages.dart';
 import 'package:provider/provider.dart';
 
 import '../../../data/models/channels/channel.dart';
 import '../../../data/models/messages/messages_provider.dart';
-import '../../../data/models/user/user.dart';
 import '../channel_messages/channel_messages.dart';
 
 class MessagesList extends StatelessWidget {
-  final User user;
+  final Query$GetAuthenticatedUser$getAuthenticatedUser user;
```

Comment:
Would the changes we need to make to classes here have implications about code reusability throughout the project? I imagine there will be places we want to display a list of users in some way, but the list of users could come from at least 2 different GraphQL Queries.

In other words, if we build a widget to show a list of Mentors which we get via the `findUsers` endpoint, can we also use that widget to display a list of Mentors which is retrieved via the  `findUserSearchById.topFoundUsers` endpoint? Both of them return a list of users. 

If there's not a clean way to do that, one side effect of this PR would be increased code duplication. If we'd need to make a `BaseMessagesList`, `FindUsersMessagesList`, and `findUserSearchByIdTopFoundUsersMessagesList` widget, that sounds like a lot of overhead.

Reply:
@m-rosario what do you think about this? Is there an easy way to deal with this, e.g. by creating a User class with constructors which can take `Query$FindUsers$user` or `Query$findUserSearchById.topFoundUsers.user` as inputs?

Reply:
Hey @andersan those are great points, here are my two cents.

If we make a generic model to store the value from different queries we would run into the same problem we had before where each query would use different fields of the models (forcing us to make non-shared fields nullable), meaning that the code that uses that model has to make a lot of safety checks.

A query should return a query result object, I think we can all agree on this. The classes that use these results can convert it into anything else they want and pass it around, however if they are consuming the data right there then why bother putting it into another object?

For your example, assuming that both queries return the exact same fields, then sure we can come up with a User class that represents a result from either query. In some of my widgets I defined auxiliary short-lived classes that make passing data around locally easier. It should be relatively easy to modify any Widget to have it take additional inputs when being constructed, but the key point is that this is done only on the classes that need it instead of having a global almighty omnipresent User class that could represent the result from one of dozens of queries.

Reply:
Thanks for addressing these questions and explaining the motivation here, Miguel. This sounds like a reasonable way to approach the different responses that will come from different GraphQL queries (indeed, even the class  `Query$GetAuthenticatedUser$getAuthenticatedUser` could have multiple different implementations depending on the amount of data which is fetched). 

The alias solution looks like an elegant way to handle this in the other case flagged by Michal.